boot/assets.go

```diff
@@ -341,27 +334,56 @@ func (o *TrustedAssetsInstallObserver) ChosenEncryptionKey(key secboot.Encryptio
 // tracking changes to the trusted boot assets and preserving managed assets,
 // provided the device model indicates this might be needed. Otherwise, nil and
 // ErrObserverNotApplicable is returned.
-func TrustedAssetsUpdateObserverForModel(model *asserts.Model) (*TrustedAssetsUpdateObserver, error) {
+func TrustedAssetsUpdateObserverForModel(model *asserts.Model, gadgetDir string) (*TrustedAssetsUpdateObserver, error) {
 	if model.Grade() == asserts.ModelGradeUnset {
 		// no need to observe updates when assets are not managed
 		return nil, ErrObserverNotApplicable
 	}
 	// trusted assets need tracking only when the system is using encryption
 	// for its data partitions
 	trackTrustedAssets := hasSealedKeys(dirs.GlobalRootDir)
-	return &TrustedAssetsUpdateObserver{
-		cache:              newTrustedAssetsCache(dirs.SnapBootAssetsDir),
-		model:              model,
-		trackTrustedAssets: trackTrustedAssets,
-	}, nil
+
+	// see what we need to observe for the run bootloader
+	runBl, runTrusted, runManaged, err := gadgetMaybeTrustedBootloaderAndAssets(gadgetDir, InitramfsUbuntuBootDir,
+		&bootloader.Options{
+			Role:        bootloader.RoleRunMode,
+			NoSlashBoot: true,
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	// and the recovery bootloader
+	seedBl, seedTrusted, seedManaged, err := gadgetMaybeTrustedBootloaderAndAssets(gadgetDir, InitramfsUbuntuSeedDir,
+		&bootloader.Options{
+			Role: bootloader.RoleRecovery,
+		})
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
if the various *Assets are empty (also based on trackTrustedAssets) we should return ErrObserverNotApplicable ?